app/views/template_invocations/_output_line.html.erb

```diff
@@ -1,8 +1,6 @@
-<% @_line_counter ||= local_assigns[:line_counter] || 0 %>
-
 <%= content_tag :div, :class => 'line ' + output_line['output_type'], :data => { :timestamp => output_line['timestamp'] } do %>
   <% output_line['output'].strip.split("\n").each do |output| %>
-    <%= content_tag(:span, (@_line_counter += 1).to_s.rjust(4).gsub(' ', '&nbsp;').html_safe + ':', :class => 'counter', :title => Time.at(output_line['timestamp'])) %>
+    <%= content_tag(:span, (@line_counter += 1).to_s.rjust(4).gsub(' ', '&nbsp;').html_safe + ':', :class => 'counter', :title => (output_line['timestamp'] && Time.at(output_line['timestamp']))) %>
```

Comment:
Since the logic around the `@line_counter` is a bit more conplex now (counting on a case where timestamp might be nil), it was just logical step to move that logic into the controller. It's better for sure :)